WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -185,15 +186,22 @@ private async Task<ImmutableArray<AliceClient>> CreateRegisterAndConfirmCoinsAsy
 					var identificationMasterKey = Slip21Node.FromSeed(masterKey.ToBytes());
 					var identificationKey = identificationMasterKey.DeriveChild("SLIP-0019").DeriveChild("Ownership identification key").Key;
 
+					var timeToWait = when - DateTimeOffset.UtcNow;
+					var fixedTimeToWait = timeToWait < TimeSpan.Zero ? TimeSpan.Zero : timeToWait;
+					await Task.Delay(fixedTimeToWait, cancellationToken).ConfigureAwait(false);
 					return await AliceClient.CreateRegisterAndConfirmInputAsync(roundState, aliceArenaClient, coin, secret, identificationKey, RoundStatusUpdater, cancellationToken).ConfigureAwait(false);
 				}
-				catch (HttpRequestException)
+				catch (HttpRequestException e)
```

Comment:
nit: introduces but doesn't use `e`

actually reporting it seems out of scope for this PR so i don't mind, it's a tiny step in the right direction